File: cpp/Hash/groupAnagrams/groupAnagrams.cpp
#include<algorithm>
#include <string>
#include <unordered_map>
#include <utility>
#include <vector>

class Soultion{
    public:
        std::vector<std::vector<std::string>> groupAnagrams(std::vector<std::string>& strs){
            std::unordered_map<std::string, std::vector<std::string>> anagramMap;
            for(auto str: strs){
                auto word = str;
                std::sort(word.begin(),word.end());
                anagramMap[word].push_back(str);
            }
            std::vector<std::vector<std::string>> res;
            for(auto anagram : anagramMap){
                res.push_back(anagram.second);
            }

            return res;
        }
};

File: Python/Hash/groupAnagrams/groupAnagrams.py
def groupAnagrams(strs: list[str]) -> list[list[str]]:

    anagramMap: dict[str, list[str]] = {}

    for string in strs:
        word: str = "".join(sorted(string))

        if word not in anagramMap:
            anagramMap[word] = []
        anagramMap[word].append(string)

    res: list[list[str]] = []
    for anagram in anagramMap.values():
        res.append(anagram)

    return res
